src/libsync/configfile.cpp

```diff
@@ -371,6 +371,16 @@ QString ConfigFile::configFile()
     return configPath() + Theme::instance()->configFileName();
 }
 
+QSettings ConfigFile::makeQSettings()
+{
+    return { configFile(), QSettings::IniFormat };
+}
+
+QSettings *ConfigFile::makeQSettingsPtr()
```

Comment:
Why would we need a ptr here, and yes I know it sucks but can you split the review, changes like this can easily be reviewed on their own and it makes reviews MUCH easier.
But yes in the end you will have to rebase this one again.